### ISSUE TITLE
Index chattiness

### ISSUE THREAD

**Issue by MikeBishop** [cabo/kramdown-rfc#167]
Currently, every instance of the word gets hyperlinked (xref) and becomes an instance in the index (iref).  In the section which defines and explains the term, that understandably means the term appears a lot, but the link isn't useful since it's a link back to the current text.

It appears that placing the iref as the child of a `<section>` causes the entire section to be placed in the index, rather than the specific paragraph -- it might be good to skip generating iref/xref tags that point to the section in which they're placed, instead placing the iref directly as a child of the target section.

**Comment by cabo** [comment]
(This issue would also benefit from an example document.)

**Comment by MikeBishop** [comment]
Example is likewise [HTTP/3](https://quicwg.org/base-drafts/h3qp_auth48/rfc9114.html#malformed) (built from https://github.com/quicwg/base-drafts/blob/h3qp_auth48/rfc9114.md) where five instances of "malformed" occur across four paragraphs in the section which defines the term.  It would probably make more sense to replace those four paragraphs' appearance in the index with a reference to the section as a whole.

Other terms have similar behavior -- I'd expect it's a common pattern in sections that define and describe a term worth indexing.

**Comment by cabo** [comment]
So what we need is a way for the target of an auto-link to control auto-links within the element carrying that target.
That sounds like a rather doable piece of post-processing, which should not be too hard to prototype.

**Comment by cabo** [comment]
Please try <https://github.com/cabo/kramdown-rfc/blob/master/bin/kramdown-rfc-autolink-iref-cleanup.rb>

Usage:

``` 
ruby kramdown-rfc-autolink-iref-cleanup.rb rfc9114.xml >rfc9114-cleaned.xml
kdrfc -h rfc9114-cleaned.xml
```

If you like what you see, I might insert this into kramdown-rfc itself (so you can use it with kdwatch etc).

Unfortunately, xml2rfc doesn't appear to act on <iref primary=true, so the cleaning effect is a bit limited.
Maybe we'll need to open an issue on that next.

**Comment by cabo** [comment]
I also opened https://github.com/ietf-tools/xml2rfc/issues/756 as the auto-generated index is still nearly useless when the primary attribute is being ignored.
And yes, https://github.com/ietf-tools/xml2rfc/issues/689...

**Comment by cabo** [comment]
I left this as a separate command `kramdown-rfc-autolink-iref-cleanup.rb` for 1.6.6.
After more feedback on how useful this is, I'll look for ways to integrate this into kramdown-rfc, probably via an option in the YAML header.
Not sure we can make much progress without xml2rfc leaving Dornröschenschlaf.

**Comment by cabo** [comment]
You may want to submit an issue to the xml2rfc repo...

**Comment by cabo** [comment]
Oh, I already did...  But maybe you could comment on https://github.com/ietf-tools/xml2rfc/issues/756

**Comment by cabo** [comment]
Good.  I need to decide whether this should be an external tool, invoked by a flag on the command line of kramdown-rfc, or both.  I'm leaning to put this into kramdown-rfc, triggered by a YAML variable in the input instead of a command-line option (I could throw in an environment variable as well).

**Comment by MikeBishop** [comment]
That seems reasonable.  I personally can't envision a situation in which I _don't_ want this behavior when I'm automatically indexing a term, but others might disagree.  (And if you're explicitly indexing particular instances, that should be respected.)

**Comment by cabo** [comment]
Closed by f72472a

1.6.10:
Just set this option in the YAML header:

```
autolink-iref-cleanup: true
```

Or keep using kramdown-rfc-autolink-iref-cleanup as a separate tool — this is now properly installed with the rest of kramdown-rfc (and has lost its .rb suffix).

Note that I **had** to keep this an option — the cleanup process canonicalizes the XML, so the output is completely different from the output without the option.  (Maybe I should offer a "cleanup: canonicalize only" option as well.)

**Comment by MikeBishop** [comment]
Yeah, I noticed that part when I tried to diff against the generated XML.
